feature(Manage Drug Packing)

lomamech opened the pull request:
Pharmaceutical packaging configuration management
- Setup
- Purchase order
- Shipping management
- Stock entries and exits

closes #6811

lomamech committed:
feature(Manage Drug Packing)
Pharmaceutical packaging configuration management
- Setup
- Purchase order
- Shipping management
- Stock entries and exits

closes #6811

lomamech committed:
test: fix unit client test

lomamech committed:
Clean and sanitaze code

lomamech committed:
test: update data for stock test integration

jmcameron commented on client/src/modules/inventory/list/modals/actions.tmpl.html:
```diff
@@ -153,14 +153,22 @@
       </label>
     </div>
 
+    <div ng-if="$ctrl.item.consumable" class="checkbox">
```
Is it possible for "non consumable" items to be packaged too?

lomamech replied:
I don't think this notion could apply to services and other medical acts

jmcameron commented on client/src/modules/stock/entry/modals/StockEntryModalForm.service.js:
```diff
@@ -28,6 +28,8 @@ function StockEntryModalForm(uuid) {
     this.isValid = false;
     this.identifier = uuid();
+    this.number_box = row.number_box || null;
+    this.package_size = row.package_size || null;
```
It seems like the this uses different terms:  package vs box.  I suggest using one for both.  Eg:  this.number_packages and this.package_size. (or with "box").  "Package" may be a little better than "box" since "package" seems more generic.  I can imagine that meds might come in tubes or barrels containing multiple doses.  In that case "box" seems awkward.

lomamech replied:
I will do it

jmcameron reviewed (commented):
I took a casual look at the code and it seems fine.  
But I did have a problem and was not able to get it to work for me.
I tried (1) enabling packaging in stock settings, (2) updating the depot to support packaging, (3) editing an inventory item to enable packaging.   Then I tried to create a purchase order and could not get the dialog to define the number of boxes, size of the packaging, and unit cost of each box to work.  I could not select anything for the "Pack Size" and the [Submit] button would not work (even though it was not greyed out).

Also, I encourage updating the purchase order document to spell out the packaging details for each item that is packaged in purchase order.

lomamech committed:
Improvement: using number package instead number of box

lomamech committed:
Clean and sanitaze code

lomamech committed:
Improvement: change label and translation key

mbayopanda commented on client/src/i18n/fr/form.json:
```diff
@@ -297,6 +298,7 @@
             "AMOUNT": "Montant",
             "AMOUNT_PAID" : "Montant Payé",
             "ARTICLES":"Articles",
+            "PACKAGE_UNIT_COST": "Coût unitaire du pacquet",
```
It must be `Coût unitaire du paquet`

mbayopanda commented on client/src/i18n/fr/form.json:
```diff
@@ -952,6 +957,7 @@
             "NAME": "Entrer le nom",
             "NOTES": "Commentaire",
             "ORIGIN": "Origine",
+            "PACKAGING": "Entrer le Conditionnement du pacquet",
```
`Entrer le Conditionnement du paquet`

mbayopanda commented on client/src/i18n/fr/inventory.json:
```diff
@@ -48,6 +48,7 @@
         "INVENTORY_TYPE_INFO" : "inventory_type : le type d'inventaire tel que Article, Service, Assemblage ou Autre",
         "INVENTORY_UNIT_INFO" : "inventory_unit : les unités de l'inventaire telles que Piece, Tablet, Box, Bottle, etc",
         "IS_BROKEN" : "est endommagé",
+        "IS_COUNT_PER_CONTAINER" : "est conditionné par boîte",
```
Since it is related to inventory and it is optional, you can set it as "Peut être conditionné par paquet"

mbayopanda reviewed (commented):
Great job @lomamech, the feature is really helpful and it is working fine.

I have some comments about: 
- Adding information (popover) in the lot modal for: "Nombre des paquets", "Taille de l'emballage", "Cout unitaire du paquet", "Quantité globale" et "Coût unitaire".
- Adding information about packing such as "Pack Number" et "Pack Size" in receipts for: Stock Entry, Stock Exit and Purchase Order

lomamech committed:
improvement(Packaging)
- Add popover information in lots entry form
- Add new column in lots registry for get information about package

lomamech committed:
feature(Drug Packing)
- Addition of information concerning the packaging on the input and output documents

- Display of information concerning the packaging on the inventory batch register

- Display of information concerning packaging during stock entries from transfers

lomamech committed:
Test: Update integration test of lot

lomamech committed:
Clean and sanitaze code

lomamech committed:
Test: Clean and sanitaze code

mbayopanda reviewed (approved):
LGTM 👍🏽 

bors r+
